src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs

```diff
@@ -0,0 +1,28 @@
+#if !ANDROID && !MACCATALYST
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues
+{
+	public class Issue21916 : _IssuesUITest
+	{
+		public Issue21916(TestDevice testDevice) : base(testDevice)
+		{
+		}
+
+		public override string Issue => "Page and control Unloaded events firing on iOS when navigating to another page"; 
+		
+		[Test]
+		[Category(UITestCategories.Shell)]
+		public void Shell_Issue21916()
+		{
+			App.WaitForElement("ButtonId");
+			App.Click("ButtonId");
+			App.WaitForElement("ButtonId");
+			App.Click("ButtonId");
+			VerifyScreenshot();
```

Comment:
Pending snapshots available in the latest build.
![image](https://github.com/user-attachments/assets/e03548b2-7d2a-4c1d-8ea1-1e6f69f305ef)
Could you commit the images?

Reply:
@jsuarezruiz , I have added the pending snapshots.